python/cudf/cudf/core/column/decimal.py

```diff
@@ -62,7 +62,46 @@ def to_arrow(self):
             buffers=[mask_buf, data_buf],
         )
 
+    def _from_integer_column(
```

Comment:
Why `DecimalColumn._from_integer_column` instead of `NumericalColumn.as_decimal_column`, which is already defined?

Reply:
Yeah, you're right about this. Updating.